client/src/app/components/Layout/Layout.jsx

```diff
@@ -70,24 +84,33 @@ export class Layout extends PureComponent {
     );
   };
 
-  handleSwitchToListView = () => this.setState({ viewType: ViewType.LIST });
-
-  handleSwitchToTilesView = () => this.setState({ viewType: ViewType.TILES });
+  handleSwitchView = viewType => this.setState({ viewType });
 
   handleViewTypeChange = () => {
+    const { updateSettings } = this.props;
     const { viewType } = this.state;
 
-    return viewType === ViewType.LIST
-      ? this.handleSwitchToTilesView
-      : this.handleSwitchToListView;
+    const newViewType =
+      viewType === ViewType.LIST ? ViewType.TILES : ViewType.LIST;
+
+    const settings = {
+      viewType: newViewType
+    };
+
+    this.handleSwitchView(newViewType);
+    this.setState({ pendingForViewType: true });
+
+    updateSettings(settings)
+      .catch(() => this.handleSwitchView(viewType))
```

Comment:
why do you handling something on catch??

Reply:
We have an optimistic update here, we update state and change view instantly. And if something goes wrong we rollback changes.

Reply:
so you can run this also in finally without catch

Reply:
no,  because in `catch` we rollback the changes if something went wrong, `finally` is invoked regardless promise is resolved or reject.

Reply:
now I get IT :)